Content.Server/Chat/Systems/AutoEmoteSystem.cs

```diff
@@ -0,0 +1,115 @@
+namespace Content.Server.Chat.Systems;
+
+using Content.Shared.Chat.Prototypes;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+public sealed class AutoEmoteSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ChatSystem _chatSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AutoEmoteComponent, ComponentInit>(OnComponentInit);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var autoEmote in EntityQuery<AutoEmoteComponent>())
+        {
+            var uid = autoEmote.Owner;
+
+            if (autoEmote.NextEmoteTime > _gameTiming.CurTime)
+                continue;
+
+            foreach ((var key, var time) in autoEmote.EmoteTimers)
+            {
+                if (time > _gameTiming.CurTime)
+                    continue;
+
+                ResetTimer(uid, key, autoEmote);
+
+                var autoEmotePrototype = _prototypeManager.Index<AutoEmotePrototype>(key);
+                if (!_random.Prob(autoEmotePrototype.Chance))
+                    continue;
+
+                if (autoEmotePrototype.WithChat)
+                {
+                    _chatSystem.TryEmoteWithChat(uid, autoEmotePrototype.EmoteId, autoEmotePrototype.HiddenFromChatWindow);
+                }
+                else
+                {
+                    _chatSystem.TryEmoteWithoutChat(uid, autoEmotePrototype.EmoteId);
+                }
+            }
+        }
+    }
+
+    private void OnComponentInit(EntityUid uid, AutoEmoteComponent autoEmote, ComponentInit args)
+    {
+        // Start timers
+        foreach (var autoEmotePrototypeId in autoEmote.Emotes)
+        {
+            ResetTimer(uid, autoEmotePrototypeId, autoEmote);
+        }
+    }
```

Comment:
Init shouldn't implicitly reset these because the comp may be deserialized.

Reply:
So I want this to work not just with `AddEmote` method but also if you have the component on an entity prototype like 
```yaml
- type: AutoEmote
  emotes: ["ZombieGroan"]
```

So I need to start the timers the first time the entity is spawned, but not if it's deserialized from like a map/grid that's initialized? Do I use `MapInit` event for this instead?

Reply:
Yeah MapInit would be preferable I think.